hrjobcontract/CRM/Hrjobcontract/Upgrader.php

```diff
@@ -1072,23 +1072,19 @@ public function createInsurancePlanTypes() {
   function upgrade_1026() {
     $messages = [];
     $pensionsTableName = CRM_Hrjobcontract_BAO_HRJobPension::getTableName();
-    
+
     $this->createPensionProviderType($messages);
     $this->replacePensionProviders($pensionsTableName, $messages);
     $this->removePensionTypeOptionGroup($messages);
 
-    try {
-      CRM_Core_DAO::executeQuery("
-        ALTER TABLE {$pensionsTableName} 
-        MODIFY pension_type INT(10) unsigned,
-        ADD CONSTRAINT pension_type_contact_id_fk
-        FOREIGN KEY(pension_type)
-        REFERENCES civicrm_contact(id)
-        ON DELETE SET NULL ON UPDATE CASCADE;
-      ");
-    } catch (Exception $e) {
-      $messages[] = "Error altering table $pensionsTableName: " . $e->getMessage();
-    }
+    CRM_Core_DAO::executeQuery("
```

Comment:
I think you should also remove the exception handling from the other helper methods (createPensionProvider, replacePensionProviders and removePensionTypeOptionGroup). It will make the upgrader stop on the first error it find.

On the other hand, I wonder if all this upgrade logic should be inside a transaction. My concern is that we might have an error in the middle of the process and then the system might get in a state that is not easily recoverable (unless you go to the database and manually fix things). I don't know if API calls would work inside a transaction though.